tests/test_benchmark_runner.py

```diff
@@ -0,0 +1,320 @@
+import inspect
+from typing import Dict
+from unittest.mock import MagicMock
+
+import pytest
+from modelgauge.annotator import Annotator
+from modelgauge.annotator_registry import ANNOTATORS
+from modelgauge.annotators.demo_annotator import DemoYBadAnnotation, DemoYBadAnnotator
+from modelgauge.dependency_helper import DependencyHelper
+from modelgauge.external_data import ExternalData
+from modelgauge.load_plugins import load_plugins
+from modelgauge.prompt import TextPrompt
+from modelgauge.record_init import InitializationRecord
+from modelgauge.secret_values import RawSecrets, get_all_secrets
+from modelgauge.suts.together_client import TogetherChatRequest, TogetherChatResponse
+
+from modelbench.benchmark_runner import *
+from modelbench.hazards import HazardDefinition, HazardScore
+from modelbench.scoring import ValueEstimate
+from modelbench.suts import ModelGaugeSut
+
+# fix pytest autodiscovery issue; see https://github.com/pytest-dev/pytest/issues/12749
+for a_class in [i[1] for i in (globals().items()) if inspect.isclass(i[1])]:
+    if a_class.__name__.startswith("Test"):
+        a_class.__test__ = False
+
+
+def fake_all_secrets(value="some-value") -> RawSecrets:
+    secrets = get_all_secrets()
+    raw_secrets: Dict[str, Dict[str, str]] = {}
+    for secret in secrets:
+        if secret.scope not in raw_secrets:
+            raw_secrets[secret.scope] = {}
+        raw_secrets[secret.scope][secret.key] = value
+    return raw_secrets
+
+
+class ATest(PromptResponseTest):
+
+    def __init__(self, uid: str, items, secrets, annotator=None):
+        super().__init__(uid)
+        self.items = items
+        self.secrets = secrets
+        self.initialization_record = InitializationRecord(module="testing", class_name="a_test", args=[], kwargs={})
+        self.annotator = annotator or ANNOTATORS.make_instance("demo_annotator", secrets=self.secrets)
+
+    def get_dependencies(self) -> Mapping[str, ExternalData]:
```

Comment:
This function claims to return something, but doesn't. Is it overridden in a subclass?

Reply:
It's just there to comply with the superclass contract. I've renamed the class to AFakeTest to make it clearer that this is just a test dummy.